tools/harness-thci/OpenThread.py

```diff
@@ -462,9 +464,23 @@ def __startOpenThread(self):
                 else:
                     self.hasActiveDatasetToCommit = False
 
+            # restore whitelist if rejoin after reset
+            if self.isPowerDown and self.isWhiteListEnabled:
+                self.__enableWhiteList()
+                for addr in self.whiteList:
+                    self.addAllowMAC(addr)
```

Comment:
Here`self.whiteList` is of`set` type which guarantees the uniqueness of elements.
https://docs.python.org/2/library/sets.html